README.md

```diff
@@ -2,31 +2,322 @@
+`Rectangle` and `Circle` objects in the same array.  
+However, our array expects `Rectangle` class instances, so we can't add `Circle`
+objects into the array.  This code won't compile:
+
+```java
+//ERROR: Can't add Circle objects to the array
+Rectangle[]tables=new Rectangle[]{
+    new Rectangle(7,5),
+    new Rectangle(2,4),
+    new Rectangle(3,5),
+    new Circle(4),
+    new Circle(6)
+};
+```
+
+The solution is to
+create an **interface** named `Shape`, and change the array declaration to
+hold objects that implement the `Shape` interface. 
+
+## Defining a `Shape` interface - Code Along
+
+![interface uml](https://curriculum-content.s3.amazonaws.com/6677/pillars/interface_uml.png)
+
+We will define an interface named `Shape`
+that defines two abstract methods `getArea()` and `getCircumference()`.
+An **abstract**  method is a method with no
+implementation (i.e. no curly braces enclosing a method body).
+
+An interface is similar to a class. However, a class defines the attributes and
+behaviors of its instances, while an interface describes behaviors
+that will be implemented by one or more classes.
+
+- An interface is never instantiated and does not have constructors.
+- An interface may define one or more abstract methods.
+  - A method declared without a method body within an interface is implicitly public and abstract.
+- An interface may define constants.
+- An interface may define default and static methods.
+- A class that declares it **implements** an interface must provide
+  an implementation (i.e. a method body) for every abstract method
+  declared in the interface, or the class must also be declared as abstract.
+
+Create a new interface named `Shape` with abstract methods `getArea()`
+and `getCircumference()` as shown below.  Both methods are abstract (no method body)
+and public  by default.
+
+```java
+public interface Shape {
+    double getArea();
+    double getCircumference();
+}
+```
+
+
+Update `Rectangle` and `Circle` to declare they implement the `Shape` interface.
+Add the `@Override` annotation to the `getArea()` and `getCircumference()` methods.
+
+
+```java
+public class Rectangle implements Shape {
+    private double height, width;
+
+    public Rectangle(double height, double width) {
+        this.height = height;
+        this.width = width;
+    }
+
+    @Override
+    public double getArea() { return height * width; }
+
+    @Override
+    public double getCircumference () { return 2 * (height + width); }
+
+}
+
+```
+
+```java
+public class Circle implements Shape {
+  private double radius;
+
+  public Circle(double radius) {this.radius = radius;}
+
+  @Override
+  public double getArea() { return Math.PI * radius * radius; }
+
+  @Override
+  public double getCircumference () { return 2 * Math.PI * radius; }
+
+}
+```
+
+While the `@Override` annotation is optional, it is best practice to use it to force
+the compiler to check the method signatures match the abstract methods in the interface.
```

Comment:
Suggestion: Maybe add a sentence like "The `@Override` annotation will provide context to the methods we defined in the `Shape` interface. We'll learn more about overriding and how that works in the coming lessons."

Reply:
updated